doc/rationale.md

```diff
@@ -0,0 +1,39 @@
+# Ignition rationale
+
+Ignition is a distro-agnostic Linux provisioning utility. When a machine boots up for the first time Ignition can manipulate it in various ways to set up the machine for operations.
+
+There are some principles Ignition is designed around. This document aims to explain these principles and how they impact Ignition's design.
+
+## Ignition only runs on the first boot
+
+Ignition is a provisioning tool, not a configuration management tool. Ignition definitely encourages immutable infrastructure, wherein if a user wishes to modify a machine then they should destroy the old machine and provision a new one. This keeps the user's machines in a well known state with relatively simple tooling.
```

Comment:
"they should destroy the old machine" sounds pretty odd in general and in bare-metal context.
Perhaps "discard the old node state and re-provision the machine"?